addons/auth_signup/controllers/main.py

```diff
@@ -54,6 +55,11 @@ def web_auth_signup(self, *args, **kw):
                     _logger.error("%s", e)
                     qcontext['error'] = _("Could not create a new account.")
 
+        elif 'signup_email' in qcontext:
+            user_ids = request.env['res.users'].sudo().search([('email', '=', qcontext.get('signup_email'))])
+            if user_ids and all(u.state != 'new' for u in user_ids):
+                return request.redirect('/web/login?%s' % url_encode({'login': user_ids.login, 'redirect': '/web'}))
```

Comment:
Maybe you can do something like this ? (`user_ids.login` might crash if you have more than 1 user)

```suggestion
            user = request.env['res.users'].sudo().search([('email', '=', qcontext.get('signup_email')), ('state', '!=', 'new')], limit=1)
            if user:
                return request.redirect('/web/login?%s' % url_encode({'login': user.login, 'redirect': '/web'}))
```